src/tabs/tabs.tsx

```diff
@@ -143,33 +151,112 @@ type TabListProps = (
      * Controls the spacing between tabs
      */
     space?: Space
-}
+
+    /**
+     * The width of the tab list.
+     *
+     * - `'maxContent'`: Each tab will be as wide as its content.
+     * - `'full'`: Each tab will be as wide as the tab list.
+     *
+     * @default 'maxContent'
+     */
+    width?: 'maxContent' | 'full'
+
+    /**
+     * How to align the tabs within the tab list.
+     *
+     * @default 'start'
+     */
+    align?: ResponsiveProp<'start' | 'center' | 'end'>
+} & ObfuscatedClassName
 
 /**
  * A component used to group `<Tab>` elements together.
  */
-function TabList({ children, space, ...props }: TabListProps): React.ReactElement | null {
+function TabList({
+    children,
+    space,
+    width = 'maxContent',
+    align = 'start',
+    exceptionallySetClassName,
+    ...props
+}: TabListProps): React.ReactElement | null {
     const tabContextValue = React.useContext(TabsContext)
 
+    const [selectedTabElement, setSelectedTabElement] = React.useState<HTMLElement | null>(null)
+    const [selectedTabStyle, setSelectedTabStyle] = React.useState<React.CSSProperties>({})
+    const tabListRef = React.useRef<HTMLDivElement>(null)
+
+    const selectedId = tabContextValue?.tabStore.useState('selectedId')
+
+    React.useLayoutEffect(() => {
```

Comment:
<img width="510" alt="image" src="https://github.com/user-attachments/assets/68cf74de-3b69-45a5-b539-c32e2595e092" />

I noticed an issue with the tab's width and position when the viewport is resized due to them being calculated on render.

Reply:
Great find, I'll fix this.